### PR TITLE
changing from timing based, to sample based collectFor

### PR DESCRIPTION
**Pull request by mattmone**
talked with Sam in Engineering and he said he would prefer to get the samples _expected_ rather than stopping at the actual time.

### CODE REVIEW

**Comment by mattmone** [comment]
also Sam said he would prefer the collection time be in seconds rather than milliseconds

File: src/webVPython.js [hunk 1]
```diff
@@ -222,6 +233,15 @@ class VernierGDX extends EventTarget {
     this._resetGraph();
   }
 
+  /**
+   * a function to set the number of samples to collect based on the period and samplesPerSecond
+   * @returns {null}
+   */
+  _setSamplesToCollect() {
+    if(this._collectFor < 0) return;
+    this._samplesToCollect = Math.floor(this.samplesPerSecond * this._collectFor);
```
Comment (stocktonkincade) [inline src/webVPython.js:242]: Probably not critical, but we take a sample at t=0 so, we actually collect (this.samplesPerSecond * this._collectFor) + 1

File: src/webVPython.js [hunk 2]
```diff
@@ -301,17 +328,15 @@ class VernierGDX extends EventTarget {
...
         this.started = false;
-      }, this._collectFor);
+        return null;
     }
 
     const chartMeasurements = this.devices
       .flatMap(deviceEl => Object.values(deviceEl.chartMeasurements))
-      .filter(measurement => ![null, undefined].includes(measurement));
+      .filter(measurement => ![null, undefined].includes(measurement))
+      .map(measurement => measurement.slice(0, this._samplesToCollect+1));
```
Comment (stocktonkincade) [inline src/webVPython.js:339]: Ah! There's the +1

**Review by stocktonkincade — APPROVED** [review]
I'm lovin' it!

**Pull request merged**